src/io_util/http_client.rs

```diff
@@ -30,7 +30,14 @@ pub struct HttpClient(
 impl HttpClient {
     /// Create a new http client.
     pub fn new() -> Self {
-        HttpClient(hyper::Client::builder().build(hyper_tls::HttpsConnector::new()))
+        HttpClient(
+            hyper::Client::builder()
+                // Disable connection pool to address weird async runtime hang.
+                //
+                // ref: https://github.com/datafuselabs/opendal/issues/473
+                .pool_max_idle_per_host(0)
```

Comment:
Does it have performance issue if we turn off ?

Reply:
It does, but not very seriously.

Most object storage services (like AWS S3, OSS, COS) will use `http/2`, which has its own connection stream and doesn't need `http/1.1`'s keep-alive.

Reply:
BTW, aws s3 always returns `Connection: close`